cmake/FindMETIS.cmake

```
@@ -14,24 +14,32 @@ find_library(METIS_LIBRARY
   metis
   PATHS
   ${METIS_DIR} $ENV{METIS_DIR} ${HIOP_METIS_DIR}
+  ENV LD_LIBRARY_PATH ENV DYLD_LIBRARY_PATH
   PATH_SUFFIXES
-  lib)
+  lib64 lib)
+
+if(METIS_LIBRARY)
+  get_filename_component(METIS_LIBRARY_DIR ${METIS_LIBRARY} DIRECTORY)
+endif()
 
 find_path(METIS_INCLUDE_DIR
   NAMES
   metis.h
   PATHS
-  ${METIS_DIR} $ENV{METIS_DIR} ${HIOP_METIS_DIR}
+  ${METIS_DIR} $ENV{METIS_DIR} ${HIOP_METIS_DIR} ${METIS_LIBRARY_DIR}/..
   PATH_SUFFIXES
   include)
 
-set(METIS_INCLUDE_DIR ${METIS_INCLUDE_DIR} CACHE PATH "Path to metis.h")
-set(METIS_LIBRARY ${METIS_LIBRARY} CACHE PATH "Path to metis library")
+set(METIS_INCLUDE_DIR "${METIS_INCLUDE_DIR}" CACHE PATH "Path to metis.h")
```

Why not put this at the top of the file and leave the name empty?
```suggestion
set(METIS_INCLUDE_DIR CACHE PATH "Path to metis.h")
```
Then `find_path` command would populate this variable. Similar can be done for `METIS_LIBRARY`.